contract/tools/ui-kit-goals/batchQuery.js

```diff
@@ -0,0 +1,176 @@
+// @ts-check
+import { E } from '@endo/far';
+
+/** @typedef {'children' | 'data'} AgoricChainStoragePathKind */
+/** @template T @typedef {import('@endo/marshal').FromCapData<T>} FromCapData<T> */
+/** @template T @typedef {import('@endo/eventual-send').ERef<T>} ERef<T> */
+
+/**
+ * @param {[kind: AgoricChainStoragePathKind, item: string]} path
+ */
+export const pathToKey = path => path.join('.');
+
+/** @param {string} key */
+export const keyToPath = key => {
+  const [kind, ...rest] = key.split('.');
+  assert(kind === 'children' || kind === 'data');
+  /** @type {[kind: 'children' | 'data', item: string]} */
+  const out = [kind, rest.join('.')];
+  return out;
+};
+
+/**
+ * @template T
+ * @param {(value: string) => T} f
+ * @param {AsyncGenerator<string[], void, unknown>} chunks
+ */
+async function* mapHistory(f, chunks) {
+  for await (const chunk of chunks) {
+    if (chunk === undefined) continue;
+    for (const value of chunk.reverse()) {
+      yield f(value);
+    }
+  }
+}
+
+/**
+ * @param {ERef<import('./makeHttpClient').LCD>} lcd
+ */
+export const makeVStorage = lcd => {
```

Comment:
this isn't importable from ui-kit?

Reply:
no. `ui-kit-goals` means I hope it will be:

 - https://github.com/Agoric/ui-kit/issues/79